File: demo/src/main/java/com/example/demo/data/repositories/UserRepository.java
package com.example.demo.data.repositories;

import com.example.demo.data.modelo.UserEntity;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.ListCrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface UserRepository extends ListCrudRepository<UserEntity, UUID> {

//    @Query("select u from UserEntity u where u.id = UUID_TO_BIN(:id)")
//    @Override
//    Optional<UserEntity> findById(@Param(value = "id") UUID id);

}

File: demoKotlin/demoKotlin/backend/src/main/kotlin/org/example/backend/data/repositories/UserRepository.kt
package org.example.backend.data.repositories

import org.example.backend.data.modelo.UserEntity
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.ListCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : ListCrudRepository<UserEntity, Long>{

    @EntityGraph(attributePaths = ["roles"])
    fun findByName(name: String): UserEntity?

    @EntityGraph(attributePaths = ["roles"])
    @Query("select u from UserEntity u")
    fun findAllWithRoles(): List<UserEntity>


}
